File: source/public/source/utils/FormatInt.mjs

export const FormatInt = function(value, length) {

	let result = "";
	let chunk  = (value).toString();

	if (chunk.startsWith("-")) {
		chunk = chunk.substr(1);
	}

	if (chunk.length < length) {

		let prefix = "";

		for (let p = 0; p < length - chunk.length; p++) {
			prefix += "0";
		}

		result = prefix + chunk;

	} else {

		result = chunk;

	}

	return result;

};

File: source/utils/FormatInt.go
package utils

import "strconv"
import "strings"

func FormatInt(number int, length int) string {

	var result string

	chunk := strconv.Itoa(number)

	if strings.HasPrefix(chunk, "-") {
		chunk = chunk[1:]
	}

	if length == 0 {

		result = chunk

	} else {

		if len(chunk) < length {

			var prefix strings.Builder

			for p := 0; p < length-len(chunk); p++ {
				prefix.WriteString("0")
			}

			result = prefix.String() + chunk

		} else {

			result = chunk

		}

	}

	return result

}
